src/traits.rs

```diff
@@ -21,3 +21,15 @@ pub trait SrcCode {
     #[must_use]
     fn generate(&self) -> String;
 }
+
+/// Trait to help collecting `Vec<impl SrcCode>` into `Vec<String>` via `.generate()`
+pub trait SrcCodeVec {
+    /// Convert the current `Vec<impl SrcCode>` into `Vec<String>`
+    fn as_src_vec(&self) -> Vec<String>;
+}
+
+impl<T: SrcCode> SrcCodeVec for Vec<T> {
+    fn as_src_vec(&self) -> Vec<String> {
+        self.iter().map(|v| v.generate()).collect()
```

Comment:
Does `self.iter().map(SrcCode::generate).collect()` work here?

Reply:
It does indeed!